pandas/io/tests/test_pytables.py

```diff
@@ -927,6 +928,53 @@ def test_encoding(self):
             result = store.select('df',Term('columns=A',encoding='ascii'))
             tm.assert_frame_equal(result,expected)
 
+    def test_latin_encoding(self):
+        # issue GH10366
+
+        def _test():
+            values = [[b'E\xc9, 17', b'', b'a', b'b', b'c'],
+                      [b'E\xc9, 17', b'a', b'b', b'c'],
+                      [b'EE, 17', b'', b'a', b'b', b'c'],
+                      [b'E\xc9, 17', b'\xf8\xfc', b'a', b'b', b'c'],
+                      [b'', b'a', b'b', b'c'],
+                      [b'\xf8\xfc', b'a', b'b', b'c'],
+                      [b'A\xf8\xfc', b'', b'a', b'b', b'c'],
+                      [np.nan, b'', b'b', b'c'],
+                      [b'A\xf8\xfc', np.nan, b'', b'b', b'c']]
+
+            def _try_decode(x, encoding='latin-1'):
+                try:
+                    return x.decode(encoding)
+                except Exception:
+                    return x
+            # not sure how to remove latin-1 from code in python 2 and 3
+            values = [[_try_decode(x) for x in y] for y in values]
+
+            examples = []
+            for dtype in ['category', object]:
+                for val in values:
+                    examples.append(pandas.Series(val, dtype=dtype))
+
+            def roundtrip(s, key='data', encoding='latin-1', nan_rep=''):
+                with ensure_clean_path(self.path) as store:
+                    s.to_hdf(store, key, format='table', encoding=encoding,
+                            nan_rep=nan_rep)
+                    retr = pandas.read_hdf(store, key)
+                    s_nan = s.replace('', np.nan)
+                    assert_series_equal(s_nan, retr)
+                    return retr
+
+            for s in examples:
+                roundtrip(s)
+
+        if compat.PY2:
+            # in Python 2: TypeError: [unicode] is not implemented as a table
+            # column
```

Comment:
if u out the py2 check at the beginning of the test then u don't need to have define _test then call it
just pull the code up one level

u had it this way before iirc